projects/plugins/protect/src/js/components/firewall-header/index.jsx

```diff
@@ -118,11 +118,18 @@ const FirewallHeader = ( { status, hasRequiredPlan } ) => {
 };
 
 const ConnectedFirewallHeader = () => {
-	const { isEnabled } = useWafData();
+	const { isEnabled, isLoading } = useWafData();
 	const { hasRequiredPlan } = useProtectData();
+	const currentStatus = isEnabled ? 'on' : 'off';
 
-	// To Do: Add loading status
-	return <FirewallHeader status={ isEnabled ? 'on' : 'off' } hasRequiredPlan={ hasRequiredPlan } />;
+	return (
+		<FirewallHeader
+			status={ isLoading ? 'loading' : currentStatus }
```

Comment:
With this current logic, the "automatic firewall is being set up" message is also shown when we're turning the firewall off.